ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -85,22 +94,28 @@ where
                         .await?;
 
                         // reveal outcome of comparison
-                        sh_comp
-                            .reveal(pctx.narrow(&Step::Reveal), RecordId::from(n))
-                            .await
+                        Ok::<bool, Error>(
+                            // desc = true will flip the order of the sort
+                            Boolean::from(false ^ desc)
+                                == sh_comp
+                                    .reveal(pctx.narrow(&Step::Reveal), RecordId::from(n))
+                                    .await?,
+                        )
                     }
                 })),
             )
-            .try_collect::<Vec<_>>()
+            .try_for_each(|x| {
+                comp.push(x);
+                future::ready(Ok(()))
+            })
```

Comment:
```suggestion
            .for_each(|x| {
                comp.push(x);
                future::ready(())
            })
```

Reply:
I think I need try since x is a result (i.e. mpc multiplication can fail)

Reply:
oh that's probably right, it will return early if one of the multiplication fails.

:shipit: